examples/tree-view/src/Node/Node.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import {
+  addChild,
+  createNode,
+  deleteNode,
+  removeChild,
+  increment
+} from './treeSlice'
+
+const Node = React.memo(({ id, parentId }) => {
+  const node = useSelector(state => state.tree[id])
+  const dispatch = useDispatch()
+
+  const { childIds, counter } = node
+
+  const handleIncrementClick = () => {
+    dispatch(increment(id))
+  }
+
+  const handleAddChildClick = () => {
+    dispatch(createNode())
```

Comment:
:hand: Can we try to figure out a way to dispatch _one_ action that has both these results?  Not sure what the necessary reducer logic would be, but we do recommend trying to [dispatch fewer actions in a row](https://redux.js.org/style-guide/style-guide#avoid-dispatching-many-actions-sequentially) and have [many reducers handle one action](https://redux.js.org/style-guide/style-guide#allow-many-reducers-to-respond-to-the-same-action).

Reply:
I could wrap them in a batch. Or simplify into a single reducer for each, addChildToNode (does createNode and addChild), removeNodeByParent (does removeChild and deleteNode). If I do that, should I still keep the specific reducers for testing purposes, or remove them and change the tests.

```js
// BEFORE
removeChild(state, action) {
  const { childId, nodeId } = action.payload
  const filteredChildIds = state[nodeId].childIds.filter(
    id => id !== childId
  )
  state[nodeId].childIds = filteredChildIds
},
deleteNode(state, action) {
  const nodeId = action.payload
  const ids = getAllDescendants(state, nodeId)

  deleteMany(state, ids)
}

// AFTER
removeNodeByParent(state, action) {
  const { childId, parentId } = action.payload
  const filteredChildIds = state[parentId].childIds.filter(
    id => id !== childId
  )
  // remove child
  state[parentId].childIds = filteredChildIds

  // delete Node(s)
  const ids = getAllDescendants(state, childId)

  deleteMany(state, ids)
},
```
```js
// BEFORE
createNode: {
  reducer: (state, action) => {
    const nextId = action.payload
    state[nextId] = { id: nextId, counter: 0, childIds: [] }
  },
  prepare: () => {
    nextId++
    return { payload: 'node_' + nextId }
  }
}
addChild: {
  reducer: (state, action) => {
    const { nodeId, nextId } = action.payload
    const childId = nextId

    state[nodeId].childIds.push(childId)
    state[childId] = { id: childId, childIds: [], counter: 0 }
  },
  prepare: nodeId => {
    return { payload: { nodeId, nextId: 'node_' + nextId } }
  }
},

// AFTER
addChildToNode: {
  reducer: (state, action) => {
    // create Node
    const { nextId, id } = action.payload
    state[nextId] = { id: nextId, counter: 0, childIds: [] }

    // add child
    state[id].childIds.push(nextId)
  },
  prepare: id => {
    nextId++
    return { payload: { id, nextId: 'node_' + nextId } }
  }
},
```

Reply:
Yeah, I think I like that change.  Update tests to match whatever logic we've got in here when we're done.